presto-docs/src/main/sphinx/optimizer/history-based-optimization.rst

```diff
@@ -0,0 +1,97 @@
+restrict_history_based_optimization_to_complex_query        Enable history based optimization only for complex queries, i.e. queries with join and aggregation      True
+history_input_table_statistics_matching_threshold           When the size difference between current table and history table exceed this threshold, do not match history statistics. When value is 0, use the default value set by hbo.history-matching-threshold       0
+=========================================================== =========================================================================================================================================================================================================== ===============
+
+Examples of HBO
+---------------
+
+An example of a query plan with HBO statistics is shown below. For a plan node, the estimation statistics will show source `HistoryBasedSourceInfo` when the statistics are from HBO.
+
+.. code-block:: text
+
+         Fragment 1 [HASH]                                                                                                                                            >
+             Output layout: [orderpriority, count]                                                                                                                    >
+             Output partitioning: SINGLE []                                                                                                                           >
+             Stage Execution Strategy: UNGROUPED_EXECUTION                                                                                                            >
+             - Project[PlanNodeId 392][projectLocality = LOCAL] => [orderpriority:varchar(15), count:bigint]                                                          >
+                     Estimates: {source: HistoryBasedSourceInfo, rows: 5 (117B), cpu: ?, memory: ?, network: ?}                                                       >
+                 - Aggregate(FINAL)[orderpriority][$hashvalue][PlanNodeId 4] => [orderpriority:varchar(15), $hashvalue:bigint, count:bigint]                          >
+                         Estimates: {source: HistoryBasedSourceInfo, rows: 5 (117B), cpu: ?, memory: ?, network: ?}                                                   >
+                         count := "presto.default.count"((count_8)) (1:50)                                                                                            >
+                     - LocalExchange[PlanNodeId 354][HASH][$hashvalue] (orderpriority) => [orderpriority:varchar(15), count_8:bigint, $hashvalue:bigint]              >
+                         - RemoteSource[2] => [orderpriority:varchar(15), count_8:bigint, $hashvalue_9:bigint]
+
+    
+Optimizations using HBO
+-----------------------
+
+DetermineJoinDistributionType & DetermineSemiJoinDistributionType
+-----------------------------------------------------------------
```

Comment:
```suggestion
-------------------------------------------------------------------
```
As a separate suggestion for you to consider, it feels as though these headings could be indented within the higher-level heading `Optimizations using HBO`. The [Sections](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections) topic of the ReStructured Text Primer suggests that you could change the 

`------` formatting to `^^^^^^` to nest these headings within `Optimizations using HBO`. 

I took screenshots of the change before and after - the headings in the main text column are not affected, but you can see the difference in the Contents in the right column. 
<img width="309" alt="Screenshot 2024-03-04 at 10 10 52 AM" src="https://github.com/prestodb/presto/assets/7013443/cb1dd2b1-7126-4403-93fb-e5bb537d1b6a">
<img width="310" alt="Screenshot 2024-03-04 at 10 21 00 AM" src="https://github.com/prestodb/presto/assets/7013443/d2a850a0-0dcc-400f-983c-9c31f6723048">